diff --git a/parse.go b/parse.go
index b642b0e..5df3e6f 100644
--- a/parse.go
+++ b/parse.go
@@ -19,7 +19,13 @@ import (
 func Parse(line string) (map[string]string, error) {
 	out := map[string]string{}
 	for _, seg := range strings.Split(line, ";") {
-		kv := strings.Split(seg, "=")
+		if strings.TrimSpace(seg) == "" {
+			continue
+		}
+		kv := strings.SplitN(seg, "=", 2)
+		if len(kv) != 2 {
+			return nil, errSegment(seg)
+		}
 		out[kv[0]] = kv[1]
 	}
 	return out, nil
